modules/videoio/src/cap.cpp

```diff
@@ -589,6 +594,22 @@ bool VideoCapture::open(int index)
     return isOpened();
 }
 
+bool VideoCapture::open(int index,int apiPreference)//added
```

Comment:
as said in the bug report. this should be just
`return open(camera + apiPreference);`